### PR TITLE
ALMS 127 Generate credentials with identity pool id and generate signed headers

### PR DESCRIPTION
**Pull request by shah279**
- Generate credentials with identity pool ID
- Generate signed headers
- Test cases added for API calls

### CODE REVIEW

File: library/src/main/java/software/amazon/location/auth/data/response/GetCredentialResponse.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package software.amazon.location.auth.data.response
+
+import com.google.gson.annotations.SerializedName
+
+
+data class Credentials(
```
Comment (olegfilimonov) [inline library/src/main/java/software/amazon/location/auth/data/response/GetCredentialResponse.kt]: split to separate file

File: library/src/main/java/software/amazon/location/auth/LocationCredentialsProvider.kt [hunk 2]
```diff
@@ -82,6 +82,24 @@ class LocationCredentialsProvider {
...
+                awsKeyValueStore.put("accessKeyId", credentials.credentials.accessKeyId)
+                awsKeyValueStore.put("secretKey", credentials.credentials.secretKey)
+                awsKeyValueStore.put("sessionToken", credentials.credentials.sessionToken)
+                awsKeyValueStore.put("expiration", credentials.credentials.expiration.toString())
+                return true
+            }
+        } catch (e: Exception) {
+           e.printStackTrace()
+        }
+        return false
```
Comment (olegfilimonov) [inline library/src/main/java/software/amazon/location/auth/LocationCredentialsProvider.kt]: if credentials could not be generated, the auth call should throw an exception, not fail silently

**Review by olegfilimonov — CHANGES_REQUESTED** [review]

File: library/src/main/java/software/amazon/location/auth/data/response/GetCredentialResponse.kt (hunk 1 above)
Comment (shah279) [reply library/src/main/java/software/amazon/location/auth/data/response/GetCredentialResponse.kt]: Done

File: library/src/main/java/software/amazon/location/auth/LocationCredentialsProvider.kt (hunk 2 above)
Comment (shah279) [reply library/src/main/java/software/amazon/location/auth/LocationCredentialsProvider.kt]: Done

**Pull request merged**